File: Strivers-A2Z-DSA-Sheet/06.Bit Manipulation/1. Learn Bit Manipulation/javascript/03.Check for odd even.js
/**
QUESTION:
Given a positive integer `N`, determine whether it is odd or even.

Approach:
Check the least-significant bit via `N & 1`. If it equals 1, N is odd; otherwise, it is even.

Time Complexity: O(1)
Space Complexity: O(1)
*/

function oddEven(N) {
    return (N & 1) ? "odd" : "even";
}

module.exports = { oddEven };

File: Strivers-A2Z-DSA-Sheet/06.Bit Manipulation/1. Learn Bit Manipulation/python/03.Check for odd even.py
"""QUESTION:
Given a positive integer `n`, return "odd" if it is odd, otherwise "even".

Approach:
Inspect the least-significant bit (`n & 1`). If it equals 1, the number is odd; otherwise, even.

Time Complexity: O(1)
Space Complexity: O(1)
"""

def odd_even(n: int) -> str:
    return "odd" if (n & 1) else "even"
